packages/components/src/RadioGroup/RadioOption.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { PropsWithChildren } from "react";
+import { XOR } from "ts-xor";
+import uuid from "uuid";
+import styles from "./RadioGroup.css";
+import { Text } from "../Text";
+
+interface BaseRadioOptions {
+  readonly value: string | number;
+  readonly disabled?: boolean;
+}
+
+interface RadioOptionsWithDescription extends BaseRadioOptions {
+  label: string;
+  /**
+   * Further description of the label
+   */
+  description?: string;
+}
+
+type RadioOptionProps = XOR<
+  RadioOptionsWithDescription,
+  PropsWithChildren<BaseRadioOptions>
+>;
+
+export function RadioOption({ children }: PropsWithChildren<RadioOptionProps>) {
```

Comment:
This isn't required to fix... but this function component definition gets a bit lost among all the typescript. Thoughts on a way we could make this a bit easier to read by moving it or highlighting it with a comment?

Reply:
I have added a comment around this one.